### PR TITLE
Don't lock modeling submissions when checking for plagiarism

### PR DESCRIPTION
**Pull request by joschahenningsen**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context
When checking modeling exercises for plagiarism the submissions are locked for the user clicking through the results without them being notified. Not 100% sure, but I considered this a bug because text and programming exercises are not locked.

### Steps for Testing

1. Check a modeling exercise for plagiarism
2. Click a few comparisons on the left
3. Go to the exercises submissions overview and check that you don't have locks for any submissions

### CODE REVIEW

**Review by KiraNickel — APPROVED** [review]
Tested on ts3, works 👍🏼  I didn't have any locks for submissions after clicking through a few plagiarism comparisons

File: src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java [hunk 1]
```diff
@@ -198,29 +204,34 @@ public ModelingSubmissionResource(SubmissionRepository submissionRepository, Res
...
         // Make sure the exercise is connected to the participation in the json response
         studentParticipation.setExercise(modelingExercise);
         modelingSubmission.getParticipation().getExercise().setGradingCriteria(gradingCriteria);
 
-        // prepare modelingSubmission for response
-        modelingSubmissionService.hideDetails(modelingSubmission, user);
-        modelingSubmission.removeNotNeededResults(correctionRound, resultId);
+        if (!withoutResults) {
+            // prepare modelingSubmission for response
+            modelingSubmissionService.hideDetails(modelingSubmission, user);
```
Comment (iozbeyli) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java:232]: Shouldn't you still hide details even without the results?

**Review by iozbeyli — COMMENTED** [review]
I added a comment and I think you should add a client service test for your new method in modeling submission service.

File: src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java (hunk 1 above)
Comment (joschahenningsen) [reply src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java]: definitely, thanks!

**Comment by joschahenningsen** [comment]
> I think you should add a client service test for your new method in modeling submission service.

I modified the test in modeling-submission-viewer.component.spec.ts to test the new method. The old one is still tested in modeling-assessment-editor.component.spec.ts.

**Review by maximilianbergmann — COMMENTED** [review]
Tested on TS3. No locks are created. 
Unfortunately, I did not see any submissions in the plagiarism editor - perhaps because my submissions were too small....
Plag

**Review by willergeorg — CHANGES_REQUESTED** [review]
Tested on TS2. I faced the same problem as @maximilianbergmann with a separate exercise. But my submissions are not very big either.

**Review by willergeorg — APPROVED** [review]
Issue solved for me.

**Review by NicolasUllmann — APPROVED** [review]
Tested on Server 2. Everything worked as expected.

**Review by cremertim — APPROVED** [review]
worked on ts1

**Review by canyelten — APPROVED** [review]
Tested on TS1, works as expected.

**Review by ndricimrr — APPROVED** [review]
Tested on TS1. Working as expected 👍

**Review by ivanchimeno — APPROVED** [review]

**Review by jpbernius — APPROVED** [review]

**Pull request merged**